cmd/garm/main.go

```diff
@@ -214,13 +215,18 @@ func main() {
 
 	router := routers.NewAPIRouter(controller, jwtMiddleware, initMiddleware, instanceMiddleware, cfg.Default.EnableWebhookManagement)
 
+	// start the metrics collector
 	if cfg.Metrics.Enable {
-		slog.InfoContext(ctx, "registering prometheus metrics collectors")
-		if err := metrics.RegisterCollectors(runner); err != nil {
-			log.Fatal(err)
-		}
 		slog.InfoContext(ctx, "setting up metric routes")
 		router = routers.WithMetricsRouter(router, cfg.Metrics.DisableAuth, metricsMiddleware)
+
+		slog.InfoContext(ctx, "register metrics")
+		if err := metrics.RegisterMetrics(); err != nil {
+			log.Fatal(err)
+		}
```

Comment:
one downside of this implementation (register the metrics here instead of doing it in the `init`):

for any further metric we want to change, we have to do a e.g.
```go
if cfg.Metrics.Enable {
       metrics.InstanceCreatedCount.WithLabelValues(
		bootstrapParams.PoolID, // label: pool_id
		e.cfg.Name,             // label: provider
	).Inc()
}
```

i'm either thinking of registering the metrics, no matter if `cfg.Metrics.Enable` is set or directly drop `cfg.Metrics.Enable` and always run `garm` with `/metrics`.

WDYT @gabriel-samfira ?

Reply:
Wouldn't we just add it to the `RegisterMetrics()` function and it gets registered here?

Reply:
In essence, `RegisterMetrics()` would act like the `init()` function, but would be capable of returning an error we can potentially handle. So instead of using the `init()` function, we define our own and call it here to register the metrics.

If metrics are enabled, we register them and start the collector loop.